goss-testing/scripts/check_bgp_neighbors_established.sh

```diff
@@ -49,19 +49,47 @@ function err_exit
     exit $rc
 }
 
-kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' > $TMPFILE ||
-    err_exit 10 "ERROR: Command failed (rc=$?): kubectl -n metallb-system get cm metallb -o jsonpath='{.data.config}' > $TMPFILE"
+TOKEN=$(curl -s -k -S -d grant_type=client_credentials -d client_id=admin-client -d client_secret=`kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d` https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token | jq -r '.access_token')
+
+# check if metalLB configmap exists
+echo "Checking for cofigmap called metallb." > $TMPFILE
+kubectl -n metallb-system get cm metallb 2>&1 |grep -i error|wc -l >> $TMPFILE
```

Comment:
Although if you are grepping for "error" just as a way to see if the kubectl command failed, then this could be done much more straightforwardly:
```bash
# Set metallb_check to 0 if we cannot get the metallb configmap from Kubernetes. Set to 1 otherwise.
if kubectl -n metallb-system get cm metallb ; then
    metallb_check=1
else
    metallb_check-0
fi
```

I'm not sure if grep was chosen here because sometimes we expect the kubectl command to succeed but to still include the string "error" somewhere in its output. If that's the case, then my previous comment is more apporpriate.

Reply:
i need to check for the error as that particular configmap only exists in csm-1.2+

Reply:
So you are saying there is a case where you would see the error string but the kubectl command would return a 0 exit code (meaning success)? Usually kubectl get exits non-0 if you try to retrieve something that doesn't exist, I think.